### PR TITLE
fix: fixing missing token when call sdk methods using Promises

### PR DESCRIPTION
**Pull request by pedroharbs**
Fixing missing token problem when call SDK methods with multiple promises, reported by @peternic.

### CODE REVIEW

File: packages/core/src/baseCogniteClient.ts [hunk 1]
```diff
@@ -147,11 +145,18 @@ export default class BaseCogniteClient {
...
+        API_KEY_HEADER,
+        options.getToken().toString()
+      );
+    }
   }
 
   public authenticate: () => Promise<string | undefined> = async () => {
     try {
-      const token = await this.getToken();
+      const token = this.getToken;
```
Comment (cognite-ornellas) [inline packages/core/src/baseCogniteClient.ts]: shouldn't be a method call?

Comment (janisharbs) [reply packages/core/src/baseCogniteClient.ts]: I removed the necessity to be a method. We aren't receiving a Promise now, just a string. If you wanna we can connect to discuss more regarding it. The tests are passing, and I did some tests with the reported scenario and now is working. @cognite-ornellas

File: packages/core/src/baseCogniteClient.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export interface ClientOptions {
   /** URL to Cognite cluster, e.g 'https://greenfield.cognitedata.com' **/
   baseUrl?: string;
   project: string;
-  getToken: () => Promise<string>;
```
Comment (roligheten) [inline packages/core/src/baseCogniteClient.ts]: How does this work with the "OIDC authentication using code authorization w pkce" example? Afaik for token refresh to work the getToken needs to be able to resolve a new token if the old one has expired. Without promises here I don't think it is possible to write a single function that can resolve a new access token externally as the function must resolve immediately. 

Case where this is important: Long-lived clients authenticated via client credentials such as APIs

**Review by cognite-ornellas — APPROVED** [review]

File: packages/core/src/baseCogniteClient.ts (hunk 2 above)
Comment (janisharbs) [reply packages/core/src/baseCogniteClient.ts]: You're right, but, the problem are exactly here when you try to get logged using api-keys using only promises without async/await. I did solve this problem on the new feature that will be on the next release, using the new auth-wraper that will do automatically the token refresh and of course the api key on this way too. Maybe we can discuss more about in a meeting, please text-me on slack and we can schedule.

**Review by cognite-ornellas — APPROVED** [review]

**Review by cognite-ornellas — APPROVED** [review]

**Pull request merged**